Fix non-unified build using InlineCallFrameMalloc in InlineCallFrameSet

ddegazio opened the pull request:
#### 05aaf7c79b6118cdb0b0cebbf23ccd4196d0e2d1
<pre>
Fix non-unified build using InlineCallFrameMalloc in InlineCallFrameSet
<a href="https://bugs.webkit.org/show_bug.cgi?id=266662">https://bugs.webkit.org/show_bug.cgi?id=266662</a>
<a href="https://rdar.apple.com/119894558">rdar://119894558</a>

Reviewed by Mark Lam.

Re-adds heap identifier definition to InlineCallFrame, and includes
InlineCallFrame.h in InlineCallFrameSet.h so we can use InlineCallFrameMalloc
in the frame set&apos;s WTF::Bag member. Also removes wtf/Bag.cpp since the
previous BagNode heap identifier is no longer declared or used.

* Source/JavaScriptCore/bytecode/InlineCallFrame.cpp:
* Source/JavaScriptCore/bytecode/InlineCallFrameSet.h:
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/Bag.cpp: Removed.
* Source/WTF/wtf/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/272383@main">https://commits.webkit.org/272383@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/30b1637a9003e08fa7667cce8f8c6cc6d66d88e0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31297 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9967 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32990 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33797 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28386 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12310 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7219 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28031 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31633 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8400 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27952 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7215 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7386 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27852 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35140 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/26862 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28464 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28306 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33519 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/31362 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7444 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5484 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31359 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9119 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/37813 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7390 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8143 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8071 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7967 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian reviewed (approved):
r=me if EWS is green.

ddegazio committed:
Fix non-unified build using InlineCallFrameMalloc in InlineCallFrameSet
https://bugs.webkit.org/show_bug.cgi?id=266662
rdar://119894558

Reviewed by Mark Lam.

Re-adds heap identifier definition to InlineCallFrame, and includes
InlineCallFrame.h in InlineCallFrameSet.h so we can use InlineCallFrameMalloc
in the frame set's WTF::Bag member. Also removes wtf/Bag.cpp since the
previous BagNode heap identifier is no longer declared or used.

* Source/JavaScriptCore/bytecode/InlineCallFrame.cpp:
* Source/JavaScriptCore/bytecode/InlineCallFrameSet.h:
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/Bag.cpp: Removed.
* Source/WTF/wtf/CMakeLists.txt:

Canonical link: https://commits.webkit.org/272383@main
